### PR TITLE
Creates an endpoint for /api/articles and tests

### PR DESCRIPTION
**Pull request by JonnyCodeMaster**
Creates an endpoint for /api/articles and creates tests for using paramteric endpoints for article_id, including error handling tests.

### CODE REVIEW

File: db/seeds/seed.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ const seed = ({ topicData, userData, articleData, commentData }) => {
     .then(() => {
       const formattedArticleData = articleData.map(convertTimestampToDate);
       const insertArticlesQueryStr = format(
-        'INSERT INTO articles (title, topic, author, body, created_at, votes, article_img_url) VALUES %L RETURNING *;',
+        "INSERT INTO articles (title, topic, author, body, created_at, votes, article_img_url) VALUES %L RETURNING *;",
```
Comment (MKD01) [inline db/seeds/seed.js:83]: Be careful not to make and push accidental changes in unrelated files

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      .then(({ body }) => {
+        expect(body.msg).toBe("Resource Not Found");
+      });
+  });
+});
+
+describe("GET /api/articles/:article_id", () => {
+  test("200 - accepts an article_id and responds with the correct article information", () => {
+    return request(app)
+      .get("/api/articles?article_id=1")
```
Comment (MKD01) [inline __tests__/app.test.js:65]: This is not the correct implementation for a parametric endpoint. The url should be /api/articles/1

File: 2. models/articles.model.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const db = require("../db/connection");
+
+exports.selectArticles = (article_id) => {
+  let sqlQuery = "SELECT articles.author, articles.title, articles.article_id, articles.body, articles.topic, articles.created_at, articles.votes, articles.article_img_url FROM articles WHERE articles.article_id = $1";
+  const queryValues = [article_id];
+
+  sqlQuery += ";";
```
Comment (MKD01) [inline 2. models/articles.model.js:7]: Not sure why this is needed to be done separately at the end?

File: 1. controllers/articles.controller.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const { selectArticles } = require("../2. models/articles.model");
+
+exports.getArticles = (req, res, next) => {
+  const { article_id } = req.query;
+
+  if (!article_id || isNaN(article_id)) {
+    return res.status(400).send({ msg: "Bad Request" });
+  }
```
Comment (MKD01) [inline 1. controllers/articles.controller.js:8]: Your controller should not be responsible for handling errors. This can and should also be handled by psql.

File: 1. controllers/articles.controller.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+  selectArticles(article_id)
+    .then((articles) => {
+      if (articles.length === 0) {
+        return res.status(404).send({ msg: "Resource Not Found" });
+      }
```
Comment (MKD01) [inline 1. controllers/articles.controller.js:14]: Same as the above this should not be handled in the controller.

File: __tests__/app.test.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  });
+
+  test("400 - responds with 'Bad Request' when an article_id is not entered", () => {
+    return request(app)
+      .get("/api/articles?article_id=")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
```
Comment (MKD01) [inline __tests__/app.test.js:90]: This would not be a valid test if this was done using a parametric endpoint.

**Review by MKD01 — COMMENTED** [review]
Please make the requested changes and resubmit this pr.

File: 2. models/articles.model.js (hunk 3 above)
Comment (JonnyCodeMaster) [reply 2. models/articles.model.js:7]: corrected

File: db/seeds/seed.js (hunk 1 above)
Comment (JonnyCodeMaster) [reply db/seeds/seed.js:83]: Completed

File: 1. controllers/articles.controller.js (hunk 4 above)
Comment (JonnyCodeMaster) [reply 1. controllers/articles.controller.js:8]: Completed

File: __tests__/app.test.js (hunk 6 above)
Comment (JonnyCodeMaster) [reply __tests__/app.test.js:90]: Completed

File: 1. controllers/articles.controller.js (hunk 5 above)
Comment (JonnyCodeMaster) [reply 1. controllers/articles.controller.js:14]: Completed

File: __tests__/app.test.js (hunk 2 above)
Comment (JonnyCodeMaster) [reply __tests__/app.test.js:65]: Completed
